hadoop/util.sh

```diff
@@ -73,3 +73,45 @@ function use_java8() {
    restart_worker_services
  fi
 }
+
+# Wait until namenode is out of safe mode. Takes 2 optional params
+# first : Number of attempts function will make to get namenode out of safemode. Default is 50
+# second : Number of seconds each attempt will sleep for waiting for namenode to come out of sleep mode. Default is 5 sec
+function wait_until_namenode_out_of_safe() {
+    n=0
+    num_attempts=$1
+    sleep_sec=$2
+    
+    if [ -z "$num_attempts" ]; then
+      num_attempts=50
+    fi
+
+    if [ -z "$sleep_sec" ]; then
+      sleep_sec=5
+    fi
+
+    nn_out_of_safe_mode=0
+    until [ $n -ge $attempts ]
+    do
+        n=$[$n+1]
+        safe_mode_stat=`hadoop dfsadmin -safemode get|awk '{print $4}'`
+        if [[ $safe_mode_stat = "ON" ]]; then
+            sudo -u hdfs hadoop dfsadmin -safemode leave
+            echo "Attempt $n/$attempts"
+            sleep $sleep_sec
+        else
+            echo "NN is out of safemode..."
+            nn_out_of_safe_mode=1
+            break
+        fi
+    done
+    if [[ $nn_out_of_safe_mode -eq 0 ]]; then
+        safe_mode_stat=`sudo -u hdfs hadoop dfsadmin -safemode get|awk '{print $4}'`
```

Comment:
Why do we have this additional check after the loop?

Reply:
To Check if the last attempt made was successful to bring namenode out of safemode or not